lib/validation.tsx

```diff
@@ -52,3 +69,41 @@ export const getErrorList = (
   }
   return errorList && errorList.length ? <ol className="gc-ordered-list">{errorList}</ol> : null;
 };
+
+const getRegexByType = (type: string | undefined, t: TFunction, value?: string) => {
+  interface RegexProps {
+    [key: string]: {
+      regex: RegExp | null;
+      error: string;
+    };
+  }
+
+  const REGEX_CONFIG: RegexProps = {
+    email: {
+      regex: /^[a-zA-Z0-9.!#$%&'*+/=?^_`{|}~-]+@[a-zA-Z0-9-]+(?:\.[a-zA-Z0-9-]+)*$/,
+      error: t("input-validation.email"),
+    },
+    text: {
+      regex: /^[0-9a-zA-Z]+$/,
+      error: t("input-validation.text"),
+    },
+    number: {
+      regex: /^[0-9]+$/,
```

Comment:
Maybe we should add `,` and `.` as valid characters as well given some people use `,` to separate every 10<sup>3</sup> and `.` due to floats.

Reply:
Yes, good idea I updated the regex for text and number.